hazelcast/src/main/java/com/hazelcast/jet/impl/processor/GroupP.java

```diff
@@ -70,12 +74,23 @@ public GroupP(
         this(singletonList(groupKeyFn), aggrOp, mapToOutputFn);
     }
 
+    @Override
+    protected void init(@NotNull Processor.Context context) throws Exception {
+        maxEntries = context.maxProcessorAccumulatedRecords();
+    }
+
     @Override
     @SuppressWarnings("unchecked")
     protected boolean tryProcess(int ordinal, @Nonnull Object item) {
         Function<Object, ? extends K> keyFn = (Function<Object, ? extends K>) groupKeyFns.get(ordinal);
         K key = keyFn.apply(item);
-        A acc = keyToAcc.computeIfAbsent(key, k -> aggrOp.createFn().get());
+        A acc = keyToAcc.computeIfAbsent(key, k -> {
+            if (keyToAcc.size() == maxEntries) {
```

Comment:
Why is this comparing the number of keys? Shouldn't it rather compare number of accumulated values?

Also some aggregation functions keep the items (e.g. collect) and some don't (e.g. counting). Shouldn't these be handled differently?

Reply:
We wanted to make it as simple as possible -> we count just entries, irrelevant of their sizes. To solve all the issues we would need to track each and every object size on a byte level (even single object with i.e. `minBy()` can take a lot of space).
Moreover, from what I have learnt `collecting` aggregations are not that common.